shared/src/components/tabulator/DateTimePickerEditor.vue

```diff
@@ -0,0 +1,218 @@
+<template>
+  <div>
+    <date-picker
+      v-if="this.typeEditorActive"
+      :type="datePickerType"
+      :clearable="false"
+      :confirm="true"
+      :open.sync="typeEditorActive"
+      ref="datepicker"
+      v-model="datePickerValue"
+      prefix-class="bkdates"
+      confirm-text="click to confirm new selection"
+      @confirm="submit"
+    />
+    <input
```

Comment:
So we don't want to re-implement the whole nullable editor inside the DateTimePickerEditor.

Have you used Vue's slots before?  We should add a slot to `NullableInput`, like this:

```vue
<div>
    <input
      class="nullible-input"
      :placeholder="smartPlaceholder"
      ref="input"
      type="text"
      v-model="value"
      @blur.prevent="submit"
      @change.prevent="submit"
      @keydown="keydown"
    >
    <span class="nullable-input-buttons">
      <slot name="buttons" />
      <i
        class="material-icons clear"
        @mousedown.prevent.stop="clear"
        title="Nullify Value"
      >cancel</i>
    </span>
  </div>
```

Then your datetime picker can look like this:
```vue
<template>
  <NullableInputEditor>
    <template v-slot:buttons>
      <i  class="material-icons" @click.prevent="pickDate">calendar</i>
    </template>
  </NullableInputEditor>
</template>
```

Reply:
Slot has been added to handle this. Should be decently extendable.